### PR TITLE
studio: cloud exps draft

### PR DESCRIPTION
**Pull request by dberenbaum**
Cloud experiments guide in Studio. Added a new PR to replace #4584 since I think it should be part of the existing guide for running experiments instead of a new page.

This means we need to update the link in https://github.com/iterative/studio/blob/6e7d2ca4b617387c6d839fa66f1bd264f3ab832f/frontend/src/utils/externalUrls.ts#LL56C3-L56C3 to point here instead.

### CODE REVIEW

File: content/docs/studio/user-guide/projects-and-experiments/run-experiments.md [hunk 1]
```diff
@@ -1,18 +1,115 @@
...
+`Save credentials`.
+
+![](https://static.iterative.ai/img/studio/s3_remote_settings_v2.png)
+
+The credentials must have the required permissions. See details and examples for
+each provider below:
+
+<details>
+
+#### AWS
```
Comment (dberenbaum) [inline content/docs/studio/user-guide/projects-and-experiments/run-experiments.md:48]: TODO: Add details and examples for each cloud provider

File: content/docs/studio/user-guide/projects-and-experiments/run-experiments.md [hunk 2]
```diff
@@ -1,18 +1,115 @@
...
+  /doc/studio/user-guide/projects-and-experiments/configure-a-project#credentials
+[profile]: https://studio.iterative.ai/user/_/profile
+[cloud credentials]:
+  /doc/studio/user-guide/projects-and-experiments/account-management
+
+### Submit a new experiment
+
+Once you have added credentials, navigate to the project and select the Git
+commit from which you want to iterate. Click the button at the top shaped like a
+flask to open the form to run an experiment.
```
Comment (dberenbaum) [inline content/docs/studio/user-guide/projects-and-experiments/run-experiments.md:83]: Not sure if there's a way to include the icon here or better describe the button.

File: content/docs/studio/user-guide/projects-and-experiments/run-experiments.md [hunk 3]
```diff
@@ -1,18 +1,115 @@
...
+
+At the top of the form, configure the cloud instance (including the region,
+size, GPU, HDD, timeout, and whether to use a spot instance). Under the
+`Parameters` tab, optionally modify any parameters of your [DVC experiment
+pipeline]. Under the `Script` tab, you can see the commands that will be run on
+the cloud instance and optionally modify them (this is not recommended unless
+you know what you are doing since it could cause the experiment to fail). At the
+bottom of the form, click `Run Experiment` to start the experiment.
+
```
Comment (dberenbaum) [inline content/docs/studio/user-guide/projects-and-experiments/run-experiments.md:92]: TODO: Add a screenshot or gif of the form.

File: content/docs/studio/user-guide/projects-and-experiments/run-experiments.md [hunk 4]
```diff
@@ -1,18 +1,115 @@
...
+[dvc experiment pipeline]: /doc/start/experiments/experiment-pipelines
+
+### Monitor a running experiment
+
+Once you submit an experiment, the instance is created and the job script is
+run. A new row is created in the experiments table under the original Git
+commit. [Live updates to metrics and plots] generated by [DVCLive] will show up
+in this row, and you can click on the experiment name to view the status and
+output log of the running experiment task.
+
```
Comment (dberenbaum) [inline content/docs/studio/user-guide/projects-and-experiments/run-experiments.md:102]: TODO: Add a screenshot or gif showing experiment metrics updates and logs.

File: content/docs/studio/get-started.md [hunk 5]
```diff
@@ -75,25 +74,18 @@ details]).
    assignments.
 
    [models dashboard]:
-     /doc/studio/user-guide/model-registry/view-and-compare-models#models-dashboard
+     /doc/studio/user-guide/model-registry/view-models#models-dashboard
```
Comment (tapadipti) [inline content/docs/studio/get-started.md:77]: Has this url changed in a different PR? In prod, it's still `view-and-compare..`

File: content/docs/studio/get-started.md [hunk 6]
```diff
@@ -75,25 +74,18 @@ details]).
...
-     (e.g. `dev`, `testing`, `prod`)
-   - Use models:
-     [download them](/doc/studio/user-guide/model-registry/use-models) or
-     [set up CI/CD](/doc/studio/user-guide/model-registry/use-models) to publish
-     or deploy models
-
-Check out
-[`What is a model registry?`](/doc/studio/user-guide/model-registry/what-is-a-model-registry)
-to get started.
+   [open its details page](/doc/studio/user-guide/model-registry/view-models#model-details-page).
```
Comment (tapadipti) [inline content/docs/studio/get-started.md:80]: I think it's best to make MR changes in a separate PR and limit this PR for exps changes only. wdyt?

File: content/docs/studio/user-guide/projects-and-experiments/what-is-a-project.md [hunk 7]
```diff
@@ -12,7 +12,7 @@ Within a project, you can:
   have pushed to your Git repository.
 - [Visualize the experiments][visualize] using plots and trend charts.
 - [Compare experiments][compare].
-- [Run new experiments][run-experiments] by submitting new Git commits directly
+- [Run new experiments][run-experiments] on your own cloud instances directly
```
Comment (tapadipti) [inline content/docs/studio/user-guide/projects-and-experiments/what-is-a-project.md:15]:
```suggestion
- [Run new experiments][run-experiments], including on your own cloud instances, directly
```

File: content/docs/studio/user-guide/projects-and-experiments/run-experiments.md [hunk 8]
```diff
@@ -1,18 +1,115 @@
-<admon>
+# Run Experiments
+
+Using your own cloud infrastructure, you can train your model on the cloud and
```
Comment (tapadipti) [inline content/docs/studio/user-guide/projects-and-experiments/run-experiments.md:3]:
```suggestion
You can train your model on the cloud and
```

File: content/docs/studio/user-guide/projects-and-experiments/run-experiments.md [hunk 9]
```diff
@@ -1,18 +1,115 @@
...
+run experiments with different hyperparameters or different datasets.
```
Comment (tapadipti) [inline content/docs/studio/user-guide/projects-and-experiments/run-experiments.md:4]:
```suggestion
run experiments with different hyperparameters or different datasets. You can configure the experiments to run on your own cloud infrastructure.
```

**Review by tapadipti — COMMENTED** [review]
Thanks for starting this @dberenbaum. I've added a few minor comments for now. I will take up some of the TODOs and work on them by Thursday day end.

File: content/docs/studio/get-started.md (hunk 6 above)
Comment (dberenbaum) [reply content/docs/studio/get-started.md:80]: Sorry, I think this was an accident. Not sure where these changes came from. Let me figure it out and revert.

File: content/docs/studio/user-guide/projects-and-experiments/run-experiments.md (hunk 1 above)
Comment (tapadipti) [reply content/docs/studio/user-guide/projects-and-experiments/run-experiments.md:49]: IMO, we should include all details about credentials in a single place, so that users don't have to read different pages to learn about credentials for projects and cloud exps. So let's update the sections `/doc/studio/user-guide/account-management#cloud-credentials` and  `/doc/studio/user-guide/projects-and-experiments/configure-a-project#data-remotes--cloud-storage-credentials`. These sections already have most of the info, and only need to be updated to make them generic, so that they don't relate only to data remotes. I'll update those sections. Then, we can link to these sections from the cloud exps docs page.

Comment (dberenbaum) [reply content/docs/studio/user-guide/projects-and-experiments/run-experiments.md:49]: Making those other sections generic and linking there for how to add credentials makes sense. 

I think we will still need experiment-specific instructions and examples because the permissions required to run experiments differ from those needed for data remotes.

File: content/docs/studio/user-guide/projects-and-experiments/run-experiments.md (hunk 3 above)
Comment (tapadipti) [reply content/docs/studio/user-guide/projects-and-experiments/run-experiments.md:92]: I'll add this once https://github.com/iterative/studio/pull/6186 is closed.

File: content/docs/studio/user-guide/projects-and-experiments/run-experiments.md (hunk 4 above)
Comment (tapadipti) [reply content/docs/studio/user-guide/projects-and-experiments/run-experiments.md:102]: I'll add this. I have gif from the past, and if there are changes, I'll create a new one.

Comment (tapadipti) [reply content/docs/studio/user-guide/projects-and-experiments/run-experiments.md:102]: Added this. I'm using a different project from what you're using though.

Comment (dberenbaum) [reply content/docs/studio/user-guide/projects-and-experiments/run-experiments.md:102]: Feel free to replace my image if you want to keep it consistent.

**Review by daavoo — CHANGES_REQUESTED** [review]
Let's put a link indicating the current AMI used under the hood so users can debug issues:

https://github.com/iterative/studio/issues/6174#issuecomment-1587356579

Perhaps it is worth mentioning those steps in a troubleshooting section.

Don't want to be pessimistic only because I faced an error in the first project I tried, but in the GPU scenarios, I would expect CUDA mismatch failures to be common. 
I think we should probably prioritize https://github.com/iterative/studio/issues/6263

**Comment by dberenbaum** [comment]
> Let's put a link indicating the current AMI used under the hood so users can debug issues:

Thanks @daavoo! Where can we link to that shows the current AMI? Can we show this in the form itself? I assume we will want to update our default AMI sometimes, so I assume we shouldn't hardcode it in the docs, right?

Why were you unable to debug without launching your own instance manually? I wonder if instead of prioritizing a troubleshooting section, we need to prioritize an issue to be able to ssh into a failed machine and debug. WDYT?

**Comment by dberenbaum** [comment]
@tapadipti What do you need help with to get this merged?

**Comment by daavoo** [comment]
> Thanks @daavoo! Where can we link to that shows the current AMI?  I assume we will want to update our default AMI sometimes, so I assume we shouldn't hardcode it in the docs, right?

We could link to https://registry.terraform.io/providers/iterative/iterative/latest/docs/resources/task#generic-1 and try to keep that updated. In Studio we currently pass the `nvidia` alias.

>  Can we show this in the form itself?

We could. 
No strong opinion, but I think I would prefer to just put a single link to the docs page with all the info and further links.
We already show a message for the check of credentials with the right permissions and I don't really like the direction of bloating the form with more info messages.


> Why were you unable to debug without launching your own instance manually?

Maybe I am missing something but I couldn't find a way to connect to a terminated instance.
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-recover.html

> I wonder if instead of prioritizing a troubleshooting section, we need to prioritize an issue to be able to ssh into a failed machine and debug. WDYT?

Makes sense, but we won't it done for the release

File: content/docs/studio/user-guide/account-management.md [hunk 10]
```diff
@@ -82,10 +82,140 @@ In this section, you can,
...
+"s3:CreateBucket",
+"s3:DeleteBucket",
+"s3:DeleteObject",
+"s3:GetObject",
+"s3:ListBucket",
+"s3:PutObject",
+```
+
+See
+https://github.com/iterative/terraform-provider-iterative/blob/a92499539f109821c021d1efb1fb01e51f1db47f/docs/guides/permissions/aws/main.tf
```
Comment (tapadipti) [inline content/docs/studio/user-guide/account-management.md:156]: @daavoo What would be the best way to list required permissions? Atm, I've listed everything I found in this file for AWS and intend to remove the filename mentioned here. I did the same thing for GCP. Do we have a more user-friendly list? If not, is the current list good enough or should I just point users to the file mentioned here?

**Comment by tapadipti** [comment]
> @tapadipti What do you need help with to get this merged?

I've made the remaining changes. Want to get a reply on a [comment about the credentials section](https://github.com/iterative/dvc.org/pull/4589#discussion_r1227951977) before merging. And add any changes if needed for the [AMI troubleshooting conversation](https://github.com/iterative/dvc.org/pull/4589#issuecomment-1588766139).

**Comment by tapadipti** [comment]
>> Let's put a link indicating the current AMI used under the hood so users can debug issues:
>
> Thanks @daavoo! Where can we link to that shows the current AMI? Can we show this in the form itself? I assume we will want to update our default AMI sometimes, so I assume we shouldn't hardcode it in the docs, right?

I second adding this to the form for now. I think there are many more issues users can encounter, and a very incomplete troubleshooting section wouldn't be good.

File: content/docs/studio/user-guide/account-management.md (hunk 10 above)
Comment (daavoo) [reply content/docs/studio/user-guide/account-management.md:156]: > Do we have a more user-friendly list? If not, is the current list good enough or should I just point users to the file mentioned here?

I don't think we have a better way for now so I am ok with leaving that

File: content/docs/studio/user-guide/account-management.md [hunk 11]
```diff
@@ -82,10 +82,140 @@ In this section, you can,
...
+Select the cloud provider (note that
+[cloud experiments](/doc/studio/user-guide/projects-and-experiments/run-experiments#cloud-experiments)
+currently support AWS and GCP). Depending on the provider, you will be asked for
+more details.
+
+![](https://static.iterative.ai/img/studio/s3_remote_settings_v2.png)
+
+The credentials must have the required permissions. For
+[cloud experiments](/doc/studio/user-guide/projects-and-experiments/run-experiments#cloud-experiments),
+the following permissions are needed:
```
Comment (dberenbaum) [inline content/docs/studio/user-guide/account-management.md:105]: Should we also link to https://dvc.org/doc/user-guide/data-management/remote-storage for data remotes?

File: content/docs/studio/user-guide/account-management.md [hunk 12]
```diff
@@ -82,10 +82,140 @@ In this section, you can,
...
+
+<details>
+
```
Comment (dberenbaum) [inline content/docs/studio/user-guide/account-management.md:108]: Not a blocker, but we should follow up to make this more user-friendly. We can include the credentials video from @dacbd and mention like he does that an easy way to get started is to give full access to ec2 and s3. We could also provide an example policy that grants all the permissions below so people can copy and paste it.

I'm less familiar with GCP/Azure, but we should be able to do something similar.

File: content/docs/studio/user-guide/account-management.md (hunk 11 above)
Comment (tapadipti) [reply content/docs/studio/user-guide/account-management.md:105]: [Added](https://github.com/iterative/dvc.org/pull/4589/commits/e6fdd788a9c837b30f7efe349f7083a01fa823b9).

File: content/docs/studio/user-guide/account-management.md (hunk 12 above)
Comment (tapadipti) [reply content/docs/studio/user-guide/account-management.md:108]: Create [an issue](https://github.com/iterative/dvc.org/issues/4613). I'll try to work on it tomorrow.

**Comment by dberenbaum** [comment]
@tapadipti Thanks for the updates! No blockers from me. Since I opened the PR, once you and/or @daavoo approve, feel free to merge tomorrow.

**Review by tapadipti — APPROVED** [review]
